### ISSUE TITLE
Opm-material PR #138 breaks opm-upscaling build

### ISSUE THREAD

**Issue by bska** [OPM/opm-upscaling#181]
Commit OPM/opm-material@d4c220e (PR OPM/opm-material#138) turns the `ParameterCace` of each `FluidSystem` into a template that depends on an `Evaluation`.  This breaks the build of [`examples/co2_blackoil_pvt.cpp`](https://github.com/OPM/opm-upscaling/blob/master/examples/co2_blackoil_pvt.cpp) and [`examples/sim_c02_impes.cpp`](https://github.com/OPM/opm-upscaling/blob/master/examples/sim_co2_impes.cpp) and basically everything that uses the opm-material dependent file [`opm/porsol/blackoil/co2fluid/BlackoilCo2Fluid.hpp`](https://github.com/OPM/opm-upscaling/blob/master/opm/porsol/blackoil/co2fluid/BlackoilCo2PVT.hpp)

The quick work-around that restores the build of opm-upscaling is to apply the patch below (i.e., to disable the build of examples that depend on opm-material's `FluidSystem` implementation).  That may even be the correct solution--at least if we do not foresee future development of this part of opm-upscaling.  The other way is to bring `BlackoilCo2PVT.hpp` up to date with the API-update from opm-material.

---

```diff
diff --git a/CMakeLists_files.cmake b/CMakeLists_files.cmake
index bbb4baf..0bde818 100644
--- a/CMakeLists_files.cmake
+++ b/CMakeLists_files.cmake
@@ -82,7 +82,7 @@ list (APPEND TEST_DATA_FILES
 list (APPEND EXAMPLE_SOURCE_FILES
        examples/aniso_implicitcap_test.cpp
        examples/aniso_simulator_test.cpp
-       examples/co2_blackoil_pvt.cpp
+       # examples/co2_blackoil_pvt.cpp
        examples/cpchop.cpp
        examples/cpchop_depthtrend.cpp
        examples/cpregularize.cpp
@@ -94,7 +94,7 @@ list (APPEND EXAMPLE_SOURCE_FILES
        examples/mimetic_periodic_test.cpp
        examples/mimetic_solver_test.cpp
        examples/sim_blackoil_impes.cpp
-       examples/sim_co2_impes.cpp
+       # examples/sim_co2_impes.cpp
        examples/sim_steadystate_explicit.cpp
        examples/sim_steadystate_implicit.cpp
        examples/steadystate_test_implicit.cpp
@@ -131,7 +131,7 @@ list (APPEND PROGRAM_SOURCE_FILES
        examples/exp_variogram.cpp
        examples/grdecldips.cpp
        examples/sim_blackoil_impes.cpp
-       examples/sim_co2_impes.cpp
+       # examples/sim_co2_impes.cpp
        examples/steadystate_test_implicit.cpp
        examples/upscale_avg.cpp
        examples/upscale_cap.cpp
```

**Comment by akva2** [comment]
unless i'm told otherwise by tptb (@alfbr) we should fix it.

**Comment by atgeirr** [comment]
@andlaus  I think you are best positioned to evaluate/fix: what would the time required to fix this be? If < 1 hour we just should do it I think, otherwise we should evaluate it more carefully.

**Comment by akva2** [comment]
i believe https://github.com/OPM/opm-upscaling/pull/182 is the correct fix.

**Comment by bska** [comment]
> i believe #182 is the correct fix.

Well, with the exception that `FluidSystem` does not depend on a template parameter so you don't need the `:: template` construct--or `typename` for that matter.  If `double` really is the correct type on which to base the `ParameterCache`, then the declaration should really be
```C++
FluidSystem::ParameterCache<double> paramCache;
```
or even
```C++
auto paramCache = FluidSystem::ParameterCache<double>{};
```

**Comment by akva2** [comment]
right you are. i just copy-pasted from the changes elsewhere without thinking - many things in the air at the same time. branch has been updated.

**Comment by bska** [comment]
Closing as fixed in #182.
